CMakeLists.txt

```diff
@@ -66,7 +70,13 @@ elseif(CLR_CMAKE_PLATFORM_UNIX_TARGET_ARM64)
 elseif(CLR_CMAKE_PLATFORM_UNIX_TARGET_AMD64)
   set(CLR_CMAKE_PLATFORM_ARCH_AMD64 1)
 elseif(WIN32)
-  set(CLR_CMAKE_PLATFORM_ARCH_AMD64 1)
+  if (CMAKE_GENERATOR MATCHES ".+ Win64")
```

Comment:
Funny thing that I did exactly that initially but then changed my mind thinking it's not worth the trouble. I'll take another look at this.